context.go

```diff
@@ -0,0 +1,69 @@
+package api2go
+
+import (
+	"golang.org/x/net/context"
+	"time"
+)
+
+// APIContext api2go context for handlers, nil implementations related to Deadline and Done.
+type APIContext struct {
```

Comment:
The Context interface leaves a lot to the implementer to expose how you set a value on it. Which means the `Value` func has to match implementation. Unfortunately, that means if we want to expose people setting values in context from middlewares we need to provide that as a public struct.